test/built-ins/RegExp/regexp-modifiers/add-ignoreCase-affects-slash-upper-p.js

```diff
@@ -0,0 +1,60 @@
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+author: Ron Buckton
+description: >
+  Adding ignoreCase (`i`) modifier affects matching for `\P{}`.
+info: |
+  Runtime Semantics: CompileAtom
+  The syntax-directed operation CompileAtom takes arguments direction (forward or backward) and modifiers (a Modifiers Record) and returns a Matcher.
+
+  Atom :: `(` `?` RegularExpressionFlags `:` Disjunction `)`
+    1. Let addModifiers be the source text matched by RegularExpressionFlags.
+    2. Let removeModifiers be the empty String.
+    3. Let newModifiers be UpdateModifiers(modifiers, CodePointsToString(addModifiers), removeModifiers).
+    4. Return CompileSubpattern of Disjunction with arguments direction and newModifiers.
+
+  Atom :: `(` `?` RegularExpressionFlags `-` RegularExpressionFlags `:` Disjunction `)`
+    1. Let addModifiers be the source text matched by the first RegularExpressionFlags.
+    2. Let removeModifiers be the source text matched by the second RegularExpressionFlags.
+    3. Let newModifiers be UpdateModifiers(modifiers, CodePointsToString(addModifiers), CodePointsToString(removeModifiers)).
+    4. Return CompileSubpattern of Disjunction with arguments direction and newModifiers.
+
+  UpdateModifiers ( modifiers, add, remove )
+  The abstract operation UpdateModifiers takes arguments modifiers (a Modifiers Record), add (a String), and remove (a String) and returns a Modifiers. It performs the following steps when called:
+
+  1. Let dotAll be modifiers.[[DotAll]].
+  2. Let ignoreCase be modifiers.[[IgnoreCase]].
+  3. Let multiline be modifiers.[[Multiline]].
+  4. If add contains "s", set dotAll to true.
+  5. If add contains "i", set ignoreCase to true.
+  6. If add contains "m", set multiline to true.
+  7. If remove contains "s", set dotAll to false.
+  8. If remove contains "i", set ignoreCase to false.
+  9. If remove contains "m", set multiline to false.
+  10. Return the Modifiers Record { [[DotAll]]: dotAll, [[IgnoreCase]]: ignoreCase, [[Multiline]]: multiline }.
+
+  GetWordCharacters ( modifiers )
+  The abstract operation GetWordCharacters takes argument modifiers (a Modifiers Record) and returns a CharSet. It performs the following steps when called:
+
+  1. Let wordCharacters be the mathematical set that is the union of all sixty-three characters in "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_" (letters, numbers, and U+005F (LOW LINE) in the Unicode Basic Latin block) and all characters c for which c is not in that set but Canonicalize(c, modifiers) is.
+  2. Return wordCharacters.
+
+esid: sec-compileatom
+features: [regexp-modifiers]
+---*/
+
+var re1 = /(?i:\P{Lu})/u;
+assert(!re1.test("A"), "\\P{Lu} should not match A");
+assert(!re1.test("a"), "\\P{Lu} should not match a (ignores case)");
+assert(!re1.test("Z"), "\\P{Lu} should not match Z");
+assert(!re1.test("z"), "\\P{Lu} should not match z (ignores case)");

```

Comment:
This was a bit confusing initially (negations mixup), but I tested it with `/\P{Lu}/ui` and it seems it should be as follows, also below.

```suggestion
assert(re1.test("A"), "\\P{Lu} should match A (ignores case)");
assert(re1.test("a"), "\\P{Lu} should match a");
assert(re1.test("Z"), "\\P{Lu} should match Z (ignores case)");
assert(re1.test("z"), "\\P{Lu} should match z");
```